plugins/modules/win_psrepository_copy.ps1

```diff
@@ -0,0 +1,234 @@
+        added to the list of possible profiles returned. Public is explicitly excluded.
+        Paths reported by the registry that don't exist on the filesystem are silently skipped.
+    #>
+    [CmdletBinding()]
+    [OutputType([System.IO.DirectoryInfo])]
+    param(
+        [Parameter()]
+        [String[]]
+        $Include ,
+
+        [Parameter()]
+        [String[]]
+        $Exclude
+    )
+
+    $regPL = 'HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\ProfileList'
+
+    # note: this is a key named "Default", not the (Default) key
+    $default = Get-ItemProperty -LiteralPath $regPL | Select-Object -ExpandProperty Default
+
+    # "ProfileImagePath" is always the local side of the profile, even if roaming profiles are used
+    # This is what we want, because PSRepositories are stored in AppData/Local and don't roam
+    $profiles = (
+        @($default) +
+        (Get-ChildItem -LiteralPath $regPL | Get-ItemProperty | Select-Object -ExpandProperty ProfileImagePath)
+    ) -as [System.IO.DirectoryInfo[]]
+
+    $profiles |
+        Where-Object -Property Exists -EQ $true |
+        Select-Wildcard -Property Name -Include $Include -Exclude $Exclude
+}
+
+function Compare-Hashtable {
+    <#
+        .SYNOPSIS
+        Attempts to naively compare two hashtables by serializing them and string comparing the serialized versions
+    #>
+    [CmdletBinding()]
+    [OutputType([bool])]
+    param(
+        [Parameter(Mandatory)]
+        [hashtable]
+        $ReferenceObject ,
+
+        [Parameter(Mandatory)]
+        [hashtable]
+        $DifferenceObject ,
+
+        [Parameter()]
+        [int]
+        $Depth
+    )
+
+    if ($PSBoundParameters.ContainsKey('Depth')) {
+        $sRef = [System.Management.Automation.PSSerializer]::Serialize($ReferenceObject, $Depth)
+        $sDif = [System.Management.Automation.PSSerializer]::Serialize($DifferenceObject, $Depth)
+    }
+    else {
+        $sRef = [System.Management.Automation.PSSerializer]::Serialize($ReferenceObject)
+        $sDif = [System.Management.Automation.PSSerializer]::Serialize($DifferenceObject)
+    }
+
+    $sRef -ceq $sDif
+}
+
+# load the repositories from the source file
+try {
+    $src = $module.Params.source
+    $src_repos = Import-Clixml -LiteralPath $src -ErrorAction Stop
+}
+catch [System.IO.FileNotFoundException] {
+    $module.FailJson("The source file '$src' was not found.", $_)
+}
+catch {
+    $module.FailJson("There was an error loading the source file '$src'.", $_)
+}
+
+$profiles = Get-ProfileDirectory -Include $module.Params.profiles -Exclude $module.Params.exclude_profiles
+
+foreach ($user in $profiles) {
+    $username = $user.Name
+
+    $repo_dir = $user.FullName | Join-Path -ChildPath 'AppData\Local\Microsoft\Windows\PowerShell\PowerShellGet'
+    $repo_path = $repo_dir | Join-Path -ChildPath 'PSRepositories.xml'
+
+    if (Test-Path -LiteralPath $repo_path) {
+        $cur_repos = Import-Clixml -LiteralPath $repo_path
```

Comment:
I'm curious, can the repository clixml contain a credential/secure string if one is assigned to that repository. If so that might cause issues trying to deserialize the object as it would be encrypted using that user's credentials.

Reply:
woah that's a really good point.. You can specify a `Credential` (and `ProxyCredential`), and I can't actually imagine them handling it any other way than the default DPAPI-based encryption. Those details then wouldn't really be copy-able; I'm going to verify and at a minimum probably update the documentation. 

If it does work this way, there's not a good workaround I can even see. In a limited circumstance, if Ansible had the credentials for both the source user and the destination user, it could be possible to do this in 2 steps, but that'd be complicated.

I'll also dig into some way I can import the XML without it failing outright, that way it doesn't affect the copying of unrelated repositories with no credentials.. unfortunately the only way I can think of off the top of my head is modifying the XML before deserialization; I tried hard to avoid that, but I'll see what I can find out.

Thanks for this!

Reply:
Yea I don't know of a way to overcome this, the closest thing would be to edit out the credentials and copy it without them but I don't think we need to handle that straight away. If anything as you said it's probably just something we need to document and just live with it for now.

Reply:
> the closest thing would be to edit out the credentials and copy it without them

Yeah I was thinking about that too.. but f the credentials were needed, I suppose it wouldn't be that helpful.
In any case I'll verify and ensure it's at least explained in the docs/notes. Thanks again, great callout.

Reply:
So beyond my expectations... they DO NOT store credentials in the XML file at all. It seems they use separate credential providers for this purpose, probably at least in part to support non-Windows: https://github.com/PowerShell/PowerShellGetv2/blob/f45bfe1a66df7063aa8de10fc4f147fdc23639af/src/PowerShellGet/public/psgetfunctions/Get-CredsFromCredentialProvider.ps1

This means that this module's operations shouldn't be affected, but I'm still unsure if such a copied repository would be usable by another user. It seems to be stored in a cred provider by sourcelocation, so it could be found, but I'm not sure that cross user the creds could actually be retrieved (it may also vary depending on the methods used in that function), and I don't have a working authenticated repository to check.

So probably still worth of a doc update, but likely no code updates; and to add on to that we won't be able to even tell that a given repository requires credentials in order to issue a warning (even `Get-PSRepository` doesn't show that the repo requires creds).

Will make some updates soon.

Reply:
Just added to the notes. Not much can be done programmatically to detect/defend against this, and same is true for both proxy params (not included in the XML file).